34.py

```diff
@@ -0,0 +1,31 @@
+class Solution:
+    def findIndex(self, nums: List[int], target: int, is_first: bool) -> int:
+        N = len(nums)
+        left, right = 0, N - 1
+
+        while left <= right:
+            mid = left + (right - left) // 2
+
+            if nums[mid] == target:
+                if is_first:
+                    if mid == left or nums[mid - 1] < target:
+                        return mid
+                    else:
+                        right = mid - 1
+                else:
+                    if mid == right or nums[mid + 1] > target:
+                        return mid
+                    else:
+                        left = mid + 1
+            elif nums[mid] < target:
+                left = mid + 1
+            else:
+                right = mid - 1
+
+        return -1
+
+    def searchRange(self, nums: List[int], target: int) -> List[int]:
+        first_idx = self.findIndex(nums, target, True)
```

Comment:
This part is a good design because you unify two functionalities (search left most and search right most) into a single interface `findIndex`. Polymorphism is achieved through different arguments passed in by the caller.